pkg/csi/service/vanilla/controller.go

```diff
@@ -248,21 +248,23 @@ func (c *controller) Init(config *cnsconfig.Config, version string) error {
 			for _, authMgr := range authMgrs {
 				go common.ComputeDatastoreMapForBlockVolumes(authMgr, config.Global.CSIAuthCheckIntervalInMin)
 			}
-			for _, vcconfig := range c.managers.VcenterConfigs {
-				isvSANFileServicesSupported, err := c.managers.VcenterManager.IsvSANFileServicesSupported(ctx,
-					vcconfig.Host)
-				if err != nil {
-					return logger.LogNewErrorf(log, "failed to verify if vSAN file services is supported or not for vCenter: %s. "+
-						"Error:%+v", vcconfig.Host, err)
-				}
-				if isvSANFileServicesSupported {
-					for _, authMgr := range authMgrs {
-						go common.ComputeFSEnabledClustersToDsMap(authMgr, config.Global.CSIAuthCheckIntervalInMin)
+			if !multivCenterTopologyDeployment {
+				for _, vcconfig := range c.managers.VcenterConfigs {
+					isvSANFileServicesSupported, err := c.managers.VcenterManager.IsvSANFileServicesSupported(ctx,
+						vcconfig.Host)
+					if err != nil {
+						return logger.LogNewErrorf(log, "failed to verify if vSAN file services is supported or not for vCenter: %s. "+
+							"Error:%+v", vcconfig.Host, err)
+					}
+					if isvSANFileServicesSupported {
+						for _, authMgr := range authMgrs {
+							go common.ComputeFSEnabledClustersToDsMap(authMgr, config.Global.CSIAuthCheckIntervalInMin)
+						}
 					}
 				}
 			}
 		}
-		if len(c.managers.VcenterConfigs) > 1 {
+		if multivCenterTopologyDeployment {
```

Comment:
Change looks good to me. But I have a question, where do we set "multivCenterTopologyDeployment" variable?

Reply:
here it is - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/7daa7d3c9f42e688feda9a796f93711f8839c579/pkg/csi/service/vanilla/controller.go#L214-L217